### DISCUSSION TITLE
Copy on highlight and right click to paste

### DISCUSSION THREAD

**Discussion by jzetterman** [ghostty-org/ghostty#3783] [Feature Requests, Ideas] [answered]
Copy on highlight and right click to paste are significant time savers, especially for IT professionals who are using the terminal to script and solve problems. If this is already possible via an entry in the config file that would be great, but I didn't see anything obvious in the documentation to allow me to configure it and I didn't see anyone here already discussing this. 

I have Alacritty configured to do both of these via the following TOML entries in its config file: 
```
[selection]
save_to_clipboard = true

[mouse]
bindings = [
  { mouse = "Right", action = "Paste" },
]
```
Something similar in Ghostty would be superb. Thanks for considering!

**Comment by JakeWharton** [comment]
First half is https://ghostty.org/docs/config/reference#copy-on-select

**Reply by jzetterman** [reply]
Nice! And I see middle click to paste is a default. Now just need to determine if I can customize that for right click or not.

**Comment by jzetterman** [comment]
I see this exists: https://ghostty.org/docs/config/keybind/reference#paste_from_selection

I just don't see a way to create a binding to a mouse action (rather than a keyboard action).

**Reply by 00-kat** [reply]
> I just don't see a way to create a binding to a mouse action (rather than a keyboard action).

Not currently possible, I had opened #3848 earlier.

**Reply by jzetterman** [reply]
Sweet, thank you. I think this is really important for laptop users as middle click is either not possible or not easy. Thanks!

**Comment by jessearmstrong** [comment]
I would also like to be able to paste on right click. I am fighting the urge to return to iterm for this feature.

**Comment by talentix-ch** [comment]
#Me too: I need the paste function from the "Primary Selection" (not from the "Clipboard Selection") on the Right Mouse Key as I have in Kitty (Linux). This is why currently I do not consider switching to Ghostty. Our workflows need to be consistent, regardless if we work on a Notebook which do not have a Middle Key or with a mouse. Some Notebooks implement the middle key by pressing both keys or by pressing another key in addition. But for me this is just annoying and useless.

**Comment by samintejas** [comment]
This single feature is holding me back from switching fully to Ghostty. Hope this will be implemented in a future release.

**Reply by pluiedev** [reply]
As a matter of posting etiquette please avoid posting "I need this" or "+1"-style comments. A simple upvote will communicate the same idea without notifying everyone in this thread.

**Comment by VesperQuartz** [comment]
Copy on select is not working

**Reply by 00-kat** [reply]
Open a new Discussion.

**Comment by 00-kat** [comment, marked as answer]
I don't see why this Discussion is still open given that conversation about right-click-to-paste has pretty much entirely moved to #4404 (yes, there is conversation about both copying and pasting there, despite the name), so I'm going to close this Discussion now unless there are any objections :).
